### ISSUE TITLE
Windows version could not load dynamic library 'DirectMLba106a7c621ea741d2159d8708ee581c11918380.dll'

### ISSUE THREAD

**Issue by tautomer** [microsoft/tensorflow-directml#36]
Hi all,

I just installed windows version and running the test from [here](https://docs.microsoft.com/zh-cn/windows/win32/direct3d12/gpu-tensorflow-windows#install-the-tensorflow-with-directml-package), but TF complains that `Could not load dynamic library 'DirectMLba106a7c621ea741d2159d8708ee581c11918380.dll'`.

The first time when I was installing `tensorflow-directml`, I got an error message

```text
ERROR: Could not install packages due to an EnvironmentError: [Errno 2] No such file or directory: 'C:\\Users\\Username\\AppData\\Local\\Packages\\PythonSoftwareFoundation.Python.3.7_qbz5n2kfra8p0\\LocalCache\\local-packages\\Python37\\site-packages\\tensorflow_estimator\\python\\estimator\\canned\\linear_optimizer\\python\\utils\\__pycache__\\sharded_mutable_dense_hashtable.cpython-37.pyc'
```

which didn't show up when I tried to do `pip install` again. I'm not sure if this error is related to the library issue.

The full input & output is

```text
>>> import tensorflow.compat.v1 as tf
>>> tf.enable_eager_execution(tf.ConfigProto(log_device_placement=True))
>>> print(tf.add([1.0, 2.0], [3.0, 4.0]))
2020-08-16 22:42:06.778056: I tensorflow/core/common_runtime/dml/dml_device_factory.cc:45] DirectML device enumeration: found 1 compatible adapters.
2020-08-16 22:42:06.778504: I tensorflow/core/platform/cpu_feature_guard.cc:142] Your CPU supports instructions that this TensorFlow binary was not compiled to use: AVX2
2020-08-16 22:42:06.780186: I tensorflow/core/common_runtime/dml/dml_device_factory.cc:32] DirectML: creating device on adapter 0 (Radeon RX Vega)
2020-08-16 22:42:06.854914: W tensorflow/stream_executor/platform/default/dso_loader.cc:71] Could not load dynamic library 'DirectMLba106a7c621ea741d2159d8708ee581c11918380.dll'; dlerror: DirectMLba106a7c621ea741d2159d8708ee581c11918380.dll not found
```

I tried to force-reinstall `tensorflow-directml`, but it didn't work.

Here is some additional information which could be useful.

```text
Python version: 3.7.8 (it's standalone installation from the store, not conda)
Windows version: 19042.450
GPU: Vega 56
Driver version: 20.7.1
```

**Comment by jstoecker** [comment]
The path limit could definitely be a problem, but it seems totally reasonable to expect the store version of python to work. This is something we should look into more.

**Comment by tautomer** [comment]
> The path limit could definitely be a problem, but it seems totally reasonable to expect the store version of python to work. This is something we should look into more.

Hi, Justion. The path limit was already bypassed before I tested all these.
![image](https://user-images.githubusercontent.com/15900605/93421588-bf2aa700-f87f-11ea-9088-4bcfab7320fe.png)

I just tested the store version again (fresh installation). No errors during the installation this time, but I still got the same problem when running the test code.

```
2020-09-17 00:56:15.044254: W tensorflow/stream_executor/platform/default/dso_loader.cc:71] Could not load dynamic library 'DirectML70b4b8b341c8bda5dc82ecd28a29d918c28282b0.dll'; dlerror: DirectML70b4b8b341c8bda5dc82ecd28a29d918c28282b0.dll not found
```

However, the library file definitely exists.
![image](https://user-images.githubusercontent.com/15900605/93422057-f2ba0100-f880-11ea-82a6-06989f844634.png)

**Comment by jstoecker** [comment]
Ah, thanks for confirming that. I suspect the problem is related to [DLL search order rules](https://docs.microsoft.com/en-us/windows/win32/dlls/dynamic-link-library-search-order), which are a little different for store apps. The trick we use to help TF locate the bundled DLL probably does not work for store apps. Sounds like a bug for us to fix. :)

**Comment by mcw256** [comment]
For anyone who wants to easily bypass this bug: 
Just copy this DirectML dll to your C:\Windows\System32 folder
Worked for me ;)

**Comment by PatriceVignola** [comment]
> For anyone who wants to easily bypass this bug:
> Just copy this DirectML dll to your C:\Windows\System32 folder
> Worked for me ;)

If possible, we do not recommend going with this approach since other applications and system components may depend on the default System32 dll. The DLL packaged in the pypi package is not meant to be used outside of TensorFlow. A much safer workaround at the moment would be to install a version of python that doesn't come from the store.

If you still choose to go down that road, please backup the previous DirectML.dll version in System32 before proceeding.

**Comment by jstoecker** [comment]
Quick update: this bug should be fixed in the new 1.15.4-dev201216 release. Try it out and let us know how it goes!
